src/app/dashboard/TenantDetails.js

```diff
@@ -0,0 +1,318 @@
+      setErr({
+        ...err,
+        databasename: true,
+        userid: false,
+        name: false,
+      });
+      databasename.current.focus();
+    } else {
+      setErr({
+        databasename: false,
+        userid: false,
+        name: false,
+        no: true,
+      });
+      let updated = {
+        ...tenant,
+      };
+      updateTenantData(tenant.id, updated).then(() => {});
+      //setModalShow(false);
+      toast.success("Tenant Details Update", {
+        position: "top-right",
+        autoClose: 1500,
+        hideProgressBar: false,
+        closeOnClick: true,
+        draggable: true,
+      });
+    }
+  };
+  return (
+    <>
+      <Dropdown className="d-inline-block">
+        <Dropdown.Toggle className="   btn-success " id="dropdown-basic">
+          Action
+        </Dropdown.Toggle>
+
+        <Dropdown.Menu>
+          <Dropdown.Item onClick={() => setDeleteshow(true)}>
+            Delete Tenant
+          </Dropdown.Item>
+          <Dropdown.Item onClick={() => navigate("/userlist")}>
+            Show users
+          </Dropdown.Item>
+        </Dropdown.Menu>
+      </Dropdown>
+      <Dropdown className="d-inline-block">
+        <Dropdown.Toggle className=" btn-danger " id="dropdown-basic">
+          Utilis
+        </Dropdown.Toggle>
+
+        <Dropdown.Menu>
+          <Dropdown.Item>Test</Dropdown.Item>
+          <Dropdown.Item>Set Tenant Url</Dropdown.Item>
+          <Dropdown.Item>Set InActive</Dropdown.Item>
+          <Dropdown.Item>Upload</Dropdown.Item>
+          <Dropdown.Item>Create tenant tables & data</Dropdown.Item>
+        </Dropdown.Menu>
+      </Dropdown>
+      <Modal show={deleteshow} onHide={() => setDeleteshow(false)} centered>
+        <Modal.Header closeButton>
+          <Modal.Title>Delete Tenant</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          Do You want To delete the tenant,Then whole database will be delete
+          under tenant
+        </Modal.Body>
+        <Modal.Footer>
+          <Button className="btn-danger" onClick={() => deleteTenant()}>
+            Remove
+          </Button>
+        </Modal.Footer>
+      </Modal>
+      <div className=" bg-white">
+        <Container className="m-1">
+          {/* <Card style={{ width: "500px", height: "400px" }} className="m-1 p-4"> */}
+          <h2 className="text-center pt-3 p-3">Tenant Details </h2>
+          <Form className="p-4">
+            <Row>
+              <Col md={6}>
+                <Form.Group className="mb-3">
+                  <Form.Label>Name :</Form.Label>
+
+                  <Form.Control
+                    type="text"
+                    placeholder="Enter Name"
+                    name="name"
+                    ref={name}
+                    value={tenant.name}
+                    isInvalid={err.name}
+                    isValid={err.no}
+                    disabled={!edit}
+                    onChange={(e) => {
```

Comment:
If possible try to reduce this onChange call back function , as it is repeated multiple times.